tests/Integration/Schema/Directives/ThrottleDirectiveTest.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace Tests\Integration\Schema\Directives;
+
+use Illuminate\Cache\RateLimiter;
+use Illuminate\Cache\RateLimiting\Limit;
+use Nuwave\Lighthouse\Exceptions\DirectiveException;
+use Nuwave\Lighthouse\Exceptions\RateLimitException;
+use Tests\TestCase;
+use Tests\Utils\Queries\Foo;
+
+class ThrottleDirectiveTest extends TestCase
+{
+    public function testWrongLimiterName(): void
+    {
+        $this->schema = /** @lang GraphQL */
+            '
+        type Query {
+            foo: Int @throttle(name: "test")
+        }
+        ';
+
+        $this->expectException(DirectiveException::class);
+        $this->graphQL(
+/** @lang GraphQL */ '
+        {
+            foo
+        }
+        '
+        );
+    }
+
+    public function testNamedLimiterReturnsRequest(): void
+    {
+        /** @var RateLimiter $rateLimiter */
+        $rateLimiter = $this->app->make(RateLimiter::class);
+
+        $this->schema = /** @lang GraphQL */
+            '
+        type Query {
+            foo: Int @throttle(name: "test")
+        }
+        ';
+
+        if (method_exists($rateLimiter, 'for')) {
+            $rateLimiter->for(
+                'test',
+                function () {
+                    return response('Custom response...', 429);
+                }
+            );
+        }
+
+        $this->expectException(DirectiveException::class);
+        $this->graphQL(
+/** @lang GraphQL */ '
+        {
+            foo
+        }
+        '
+        );
+    }
+
+    public function testNamedLimiter(): void
+    {
+        /** @var RateLimiter $rateLimiter */
+        $rateLimiter = $this->app->make(RateLimiter::class);
+
+        if (! method_exists($rateLimiter, 'for') || ! class_exists('Illuminate\Cache\RateLimiting\Limit')) {
+            return;
```

Comment:
You can use `$this->markTestSkipped()` and give a reason. Look around for other uses to fix the resulting PHPStan error.

Reply:
Yep, marked tests as skipped and also used `AppVersion::below(8.0)` to check version